src/Data/List/Properties.agda

```diff
@@ -106,6 +106,13 @@ map-compose : {g : B → C} {f : A → B} → map (g ∘ f) ≗ map g ∘ map f
 map-compose []       = refl
 map-compose (x ∷ xs) = cong (_ ∷_) (map-compose xs)
 
+map-injective : ∀ {f : A → B} (xs ys : List A) → (∀ {x y} → f x ≡ f y → x ≡ y) →
```

Comment:
See the discussion in #1141 where we discuss using the `Injective` predicate, and putting this proof under `Data.List.Function.Map` and then re-exporting `map-injective` publicly in this file if its deemed useful. Given that PR has been sitting there for a while, I'm happy for you to go ahead and move it there in this one.

Reply:
This `map-injective` proof uses `Data.List.Properties.∷-injective`, is this circular dependency avoidable in some way?

Reply:
I've update the type for now :)

Reply:
Hmm excellent point! Maybe it needs to be the other way around. Have `map-injective` in `Data.List.Properties` and then re-export it in `Data.List.Function.Map`...

Either that or have `Data.List.Properties.Core` which I'm personally against (though I know those sort of files have their supporters!)

Reply:
Okay sounds good. I'm happy to leave it at that! I'll wait for Travis to go green and then merge it in 👍

Reply:
I guess if this kind of properties keep pilling up it might be worth revisiting this.